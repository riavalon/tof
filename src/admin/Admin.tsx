import React from 'react'
import { BrowserRouter as Router, Route, useParams, NavLink } from 'react-router-dom'

import './Admin.styles.scss'

const AdminTopic = () => {
  const params = useParams<{ adminTopicID: string }>()
  console.log(params)
  return (
    <div>
      <h2>Admin Topic #{params.adminTopicID}</h2>
    </div>
  )
} 

export type adminProps = {
  user: Object,
}

const Admin = ({ user }: adminProps) => {
  return (
    <div className="page-container">
      <Router>
        <div className="tof-admin">
          <ul className="tof-admin__side-nav">
              <li>
                <NavLink activeClassName="active" to="/admin/posts">Posts</NavLink>
              </li>

              <li>
                <NavLink activeClassName="active" to="/admin/fungi">Fungi</NavLink>
              </li>

              <li>
                <NavLink activeClassName="active" to="/admin/motd">MotD</NavLink>
              </li>
          </ul>
          <div className="tof-admin__nav-content">
            <Route path={`/admin/:adminTopicID`} component={AdminTopic} />
          </div>
        </div>
      </Router>
    </div>
  )
}

export default Admin