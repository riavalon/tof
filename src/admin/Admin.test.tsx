import React from 'react'
import { render } from '@testing-library/react'

import Admin from './Admin'

describe('Admin Page', () => {
  const mockUser = {}

  it('should render without issue', () => {
    const { baseElement } = render(<Admin user={mockUser} />)
    expect(baseElement).toMatchSnapshot()
  })
})