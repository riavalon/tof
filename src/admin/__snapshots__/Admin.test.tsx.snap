// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page should render without issue 1`] = `
<body>
  <div>
    <div
      class="page-container"
    >
      <div
        class="tof-admin"
      >
        <ul
          class="tof-admin__side-nav"
        >
          <li>
            <a
              href="/admin/posts"
            >
              Posts
            </a>
          </li>
          <li>
            <a
              href="/admin/fungi"
            >
              Fungi
            </a>
          </li>
          <li>
            <a
              href="/admin/motd"
            >
              MotD
            </a>
          </li>
        </ul>
        <div
          class="tof-admin__nav-content"
        />
      </div>
    </div>
  </div>
</body>
`;
